### PR TITLE
Support BoldText for a11y in android 

### PR DESCRIPTION
**Pull request by hangyujin**
fixed: #86187

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/io/flutter/view/AccessibilityBridge.java [hunk 1]
```diff
@@ -539,6 +546,26 @@ private boolean shouldSetCollectionInfo(final SemanticsNode semanticsNode) {
...
+  @RequiresApi(31)
+  private void setBoldTextFlag() {
+    if (rootAccessibilityView == null || rootAccessibilityView.getResources() == null) {
+      return;
+    }
+    int fontWeightAdjustment =
+        rootAccessibilityView.getResources().getConfiguration().fontWeightAdjustment;
+    boolean shouldBold =
+        fontWeightAdjustment != Configuration.FONT_WEIGHT_ADJUSTMENT_UNDEFINED
+            && fontWeightAdjustment >= 300;
```
Comment (chunhtai) [inline shell/platform/android/io/flutter/view/AccessibilityBridge.java:559]: Why 300? is there documentation around this value? This should be a constant for readability.

**Review by chunhtai — APPROVED** [review]
LGTM, just some questions

File: shell/platform/android/io/flutter/view/AccessibilityBridge.java (hunk 1 above)
Comment (hangyujin) [reply shell/platform/android/io/flutter/view/AccessibilityBridge.java:560]: Added a comment
300 is bold - normal = w700 - w400

Comment (chunhtai) [reply shell/platform/android/io/flutter/view/AccessibilityBridge.java:560]: Can you also add a constant for the value 300?
```
static final int kBoldTextAdjustment = 300
```

**Pull request merged**